api/proto/dsse.proto

```diff
@@ -29,6 +29,10 @@ message DSSERequestV0_0_2 {
     io.intoto.Envelope envelope = 1 [(google.api.field_behavior) = REQUIRED];
     // All necessary verification material to verify all signatures embedded in the envelope
     repeated Verifier verifiers = 2 [(google.api.field_behavior) = REQUIRED];
+    // An optional hash algorithm to when recording the dsse payload hash in the log, by default the system will
+    // use SHA2_256
+    dev.sigstore.common.v1.HashAlgorithm payloadHashAlgorithm = 3 [(google.api.field_behavior) = OPTIONAL];
```

Comment:
open to other names for this

Reply:
Maybe just `algorithm` consistent with protobuf-specs? https://github.com/sigstore/protobuf-specs/blob/95810107164a2b4c76bcdad51439f61031663078/protos/sigstore_common.proto#L116

Reply:
yeah I was wondering if that would be confusing? I don't think `payloadHashAlgorithm` isn't confusing. It'd be nice if it was clear that the server is going to use this, and it doesn't affect anything else they provide in the request.

Reply:
:woman_shrugging:

Reply:
Naming isn't my strong suit, I think either would be fine. Given that it's used as part of the signature too, a more generic `algorithm` might be more accurate, but either way, they need to read the doc string.